### ISSUE TITLE
TestFactory doesn't close registered `AutoCloseable` object

### ISSUE THREAD

**Issue by Yang-33** [kotest/kotest#3738]
Kotest version: 5.7.2.
(Thank you for maintaining Kotest! I love many assertions and inspectors supported by Kotest. They are very useful!)

The TestFactory described in https://kotest.io/docs/framework/test-factories.html doesn't call the `close()` method, even when an `AutoCloseable` object is wrapped by the `autoClose` function defined in Kotest. Below is a minimal test case to reproduce this.

Is there a way to automatically call close when using TestFactory? If so, I would like to know!

``` kotlin
data class A(val text: String) : AutoCloseable {
    override fun close() {
        logger.info { "close $text" }
    }
    companion object : KLogging()
}

fun aTest() = shouldSpec {
    should("close is not called. why...?") {
        val a = autoClose(A("close is not called...")) // This |a|'s `close` is not called
        a.shouldNotBeNull()
    }
}

class Test : ShouldSpec({
    include(aTest())
    should("close is called") {
        val a = autoClose(A("close is called!!!"))
        a.shouldNotBeNull()
    }
})
```

**Comment by kshired** [comment]
@Yang-33 It will be fixed by #3739

**Comment by Yang-33** [comment]
Wow, thank you!
